Both items in this patch are due to currently unsupported facilities for OSv.
When we gain support for them, we should drop this.

In particular the socket one, it is possible to override that with a -D switch
during cmake configure. But overriding it with an empty string make the
conditional resolve to true.
--- a/storage/innobase/CMakeLists.txt	2014-07-18 19:48:39.000000000 +0400
+++ b/storage/innobase/CMakeLists.txt	2014-09-03 12:34:03.137064214 +0400
@@ -22,13 +22,13 @@ INCLUDE(CheckCSourceRuns)
 # OS tests
 IF(UNIX)
   IF(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    CHECK_INCLUDE_FILES (libaio.h HAVE_LIBAIO_H)
-    CHECK_LIBRARY_EXISTS(aio io_queue_init "" HAVE_LIBAIO)
+#    CHECK_INCLUDE_FILES (libaio.h HAVE_LIBAIO_H)
+#    CHECK_LIBRARY_EXISTS(aio io_queue_init "" HAVE_LIBAIO)
     ADD_DEFINITIONS("-DUNIV_LINUX -D_GNU_SOURCE=1")
-    IF(HAVE_LIBAIO_H AND HAVE_LIBAIO)
-      ADD_DEFINITIONS(-DLINUX_NATIVE_AIO=1)
-      LINK_LIBRARIES(aio)
-    ENDIF()
+#    IF(HAVE_LIBAIO_H AND HAVE_LIBAIO)
+#      ADD_DEFINITIONS(-DLINUX_NATIVE_AIO=1)
+#      LINK_LIBRARIES(aio)
+#    ENDIF()
   ELSEIF(CMAKE_SYSTEM_NAME MATCHES "HP*")
     ADD_DEFINITIONS("-DUNIV_HPUX")
   ELSEIF(CMAKE_SYSTEM_NAME STREQUAL "AIX")
--- a/cmake/mysql_version.cmake	2014-07-18 19:48:39.000000000 +0400
+++ b/cmake/mysql_version.cmake 2014-09-03 14:31:31.580182813 +0400
@@ -83,7 +83,7 @@ ENDIF()
 
 
 IF(NOT MYSQL_UNIX_ADDR)
-  SET(MYSQL_UNIX_ADDR "/tmp/mysql.sock")
+  SET(MYSQL_UNIX_ADDR "")
 ENDIF()
 IF(NOT COMPILATION_COMMENT)
   SET(COMPILATION_COMMENT "Source distribution")
diff -ur mysql-5.6.21/sql/sql_acl.cc mysql-5.6.21-2/sql/sql_acl.cc
--- mysql-5.6.21/sql/sql_acl.cc	2017-09-28 13:23:12.536504772 +0200
+++ mysql-5.6.21-2/sql/sql_acl.cc	2014-09-11 15:53:31.000000000 +0200
@@ -3212,7 +3212,7 @@
       * An empty password is considered to be of mysql_native type.
     */
     
-    if (combo->plugin.str == NULL || combo->plugin.str == '\0')
+    if (combo->plugin.str == NULL || combo->plugin.str[0] == '\0')
     {
       if (combo->uses_identified_by_password_clause)
       {

